File: 프로그래머스/2/64065. 튜플/튜플.js
function solution(s) {
  const result = []

  const tuples = JSON.parse(s.replace(/{/g, "[").replace(/}/g, "]")).sort(
    (a, b) => a.length - b.length
  );

  const set = new Set()

  for (let tuple of tuples) {
    for (let v of tuple) {
      if (!set.has(v)) {
        set.add(v)
        result.push(v)
      }
    }
  }
  return result
}

File: 프로그래머스/2/64065. 튜플/튜플.py
import json

def solution(s):
    result = []

    s_processed = s.replace('{', '[').replace('}', ']')
    tuples = json.loads(s_processed)
    tuples.sort(key=len)

    seen_elements = set()

    for current_tuple in tuples:
        for element in current_tuple:
            if element not in seen_elements:
                seen_elements.add(element)
                result.append(element)
    return result